src/UI/UIUtils.java

```diff
@@ -323,8 +323,18 @@ public static COSpriteSpec fromDisk(String col, String fac)
   public static final COSpriteSpec BW = COSpriteSpec.fromDisk("Brenner's"     , "Wolves");
   public static final COSpriteSpec NW = COSpriteSpec.fromDisk("New Rubinelle" , "Wolves");
   public static final COSpriteSpec LA = COSpriteSpec.fromDisk("Lazurian"      , "Army");
-  public static final COSpriteSpec IA = COSpriteSpec.fromDisk("IDS"           , "Army");
+  public static final COSpriteSpec IDS = COSpriteSpec.fromDisk("IDS"          , "Army");
 
+  public enum SourceGames
+  {
+    AW1(OS), AW2(BM), AW3(GE), AW4(IDS), DEFEND_PEACE(RT); // More? :v
```

Comment:
Right, so I defined this game enum here, and arbitrarily picked which faction they'd take their color from.
Having a game enum is sort of required to have filtering by game, and having a color for each game is nice for GUI things.

This GUI is mostly to make it easier to pick through an unreasonably large cast of COs, since that's super where we're headed (and arguably already hit on the test branch)

I figured having the ability to toggle between the filter strategies would do two important things:
You can more easily limit what you're looking at to what you care about ("I'm looking for an OS CO" vs "I only want to get AW2 COs").
It lets you navigate to specific things a little faster/easier... albeit in a way that isn't super intuitive.

I moved all the "current state" tracking logic into the view, since it has to handle wrapping concerns.
The basic structure is:
View owns a list of outer bins
Outer bins own a list of inner bins
Inner bins can have multiple lines, since the user can fiddle with the window size

Everything is re-generated in place when the filter strategy flips.
Filter strategy determines what is "outer" vs "inner" (with "inner" being the category for each row)

I'm not super thrilled with the way this divides the code (stealing from the short file to make the longer one longer), but the division of responsibilities feels better.


I guess I kind of assumed "You can see the GUI, GUI tells you what it is" would be good enough. This is a silly assumption because a lot of GUI stuff you wrote confused me when I first read it...